roles/junit2json/tasks/merge.yml

```diff
@@ -0,0 +1,40 @@
+# SPDX-License-Identifier: Apache-2.0
+---
+# tasks file merge.yml for role redhatci.ocp.junit2json
+# merges multiple XMLs into 1 and converts it into JSON
+
+- name: Generate XML files list
+  ansible.builtin.set_fact:
+    junit2json_reports_list: "{{ junit2json_reports_list | default([]) + [item] }}"
+  loop: "{{ _junit2json_found_reports.files | map(attribute='path') }}"
+
+# debug only:
+- name: Print XML reports' filenames data
+  ansible.builtin.debug:
+    msg:
+      - "XML files names: {{ junit2json_reports_list }}"
+  when:
+    - junit2json_debug is defined
+    - junit2json_debug
+
+- name: Merge multiple JUnit XML files into single consolidated report
+  ansible.builtin.shell:
```

Comment:
If just using one single command, please use `command` module instead of `shell`